backend/tournesol/admin.py

```diff
@@ -40,27 +42,30 @@ def lookups(self, request, model_admin):
         return [("", "-")] + [(v, v) for v in field_values]
 
     def queryset(self, request, queryset):
+        """Filter the queryset according to the selected metadata filter"""
         if self.value():
             json_field_query = {f"metadata__{self.metadata_key}": self.value()}
-            return queryset.filter(**json_field_query)
+            queryset = queryset.filter(**json_field_query)
         elif self.value() == "":
             json_field_query = (
                 Q(**{f"metadata__{self.metadata_key}": ""})
                 | Q(**{f"metadata__{self.metadata_key}": None})
             )
-            return queryset.filter(json_field_query)
-        else:
-            return queryset
+            queryset = queryset.filter(json_field_query)
+
+        return queryset
```

Comment:
R1705: Unnecessary "elif" after "return", remove the leading "el" from "elif" (no-else-return)

Reply:
Ah yes, it makes sense. After `return`, `elif:` should be replaced with `if:`.

To please the linter we would write:
```py
    def queryset(self, request, queryset):
        if self.value():
            json_field_query = {f"metadata__{self.metadata_key}": self.value()}
            return queryset.filter(**json_field_query)
        if self.value() == "":
            json_field_query = (
                Q(**{f"metadata__{self.metadata_key}": ""})
                | Q(**{f"metadata__{self.metadata_key}": None})
            )
            return queryset.filter(json_field_query)
        return queryset
 ```

Reply:
I don't really like that either, because the last line, "return queryset" is never used.

maybe rather:
```
    def queryset(self, request, queryset):
        if self.value():
            json_field_query = {f"metadata__{self.metadata_key}": self.value()}
            return queryset.filter(**json_field_query)

        json_field_query = (
            Q(**{f"metadata__{self.metadata_key}": ""})
            | Q(**{f"metadata__{self.metadata_key}": None})
        )
        return queryset.filter(json_field_query)
```

Reply:
> I don't really like that either, because the last line, "return queryset" is never used.

I am not sure about the implementation details of list filters in Django admin and in which cases this method is called. But, from a purely logical point of view at least, it is useful to have this default case to distinguish between `None` and empty string.

When `self.value()` is None (i.e the filter is not active), no object should be excluded. On the contrary when it's an empty string (e.g passed via admin query param `?metadataLanguage=`) it means that only entities whose metadata field is empty (or JSON `null`) should be displayed.

Reply:
ok